micropy/config/config.py

```diff
@@ -132,11 +154,27 @@ def set(self, key: str, value: Any) -> Any:
             Any: Updated config
 
         """
-        full_path = tuple(i for i in key.split('.'))
-        path = full_path[:-1]
-        p_key = full_path[-1]
+        path, target = self.parse_key(key)
         remapped = iterutils.remap(self._config, lambda p, k, v: (
-            k, value) if p == path and k == p_key else (k, v))
+            k, value) if p == path and k == target else (k, v))
         self._config = remapped
         self.log.debug(f"set config value [{key}] -> {value}")
         return self.sync()
+
+    def pop(self, key: str) -> Any:
+        """Delete and return value at key.
+
+        Args:
+            key (str): Key to pop.
+
+        Returns:
+            Any: Popped value.
+
+        """
+        path, target = self.parse_key(key)
+        value = self.get(key)
+        remapped = iterutils.remap(self._config, lambda p, k,
+                                   v: False if p == path and k == target else True)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The if expression can be replaced with 'not test'](https://app.codacy.com/app/BradenM/micropy-cli/pullRequest?prid=4593453)